packages/commonwealth/client/scripts/views/components/header/login_selector.tsx

```diff
@@ -61,86 +60,101 @@ export const LoginSelectorMenuLeft = ({
   nAccountsWithoutRole,
 }: LoginSelectorMenuLeftAttrs) => {
   const navigate = useCommonNavigate();
+  const forceRerender = useForceRerender();
 
-  const [isEditProfileModalOpen, setIsEditProfileModalOpen] = useState(false);
   const [isLoginModalOpen, setIsLoginModalOpen] = useState(false);
   const [isSelectAddressModalOpen, setIsSelectAddressModalOpen] =
     useState(false);
+  const [profileId, setProfileId] = useState(null);
+  const [selectedAddress, setSelectedAddress] = useState(null);
+
+  useEffect(() => {
+    // force rerender when new address is connected
+    app.user.isFetched.on('redraw', () => {
```

Comment:
every subscription should be unsubscribed  on component unmount => example in overview/index.ts